public/blogPosts/fall2019/js-chat-1/index.md

```diff
@@ -827,117 +864,127 @@ undefined
+		console.log(this.age);
+	},
 };
 ```
 
-When `getNameInsideA` and `getAgeInsideA` is defined, the 
+When `getNameInsideA` and `getAgeInsideA` is defined, the
 `this` is referring to the global object (not to `a`).
 Therefore, `getNameInsideA` gives `undefined` no matter
-how you call it. 
+how you call it.
 And yes, `bind`, `call`, `apply` do not work on arrow functions.
 In fact, the arrow function can be equivalently written as
 
 ```js
 const a = () => {};
 // is equivalent to
-const a = (function () {}).bind(this);
+const a = function () {}.bind(this);
 ```
 
-However, the `this` binding in arrow functions can still 
+However, the `this` binding in arrow functions can still
 be subjected to implicit/explicit runtime `this` binding.
 This will be the case when the scope also has a runtime
 bound `this`. A concrete example will be an arrow function
 defined in a function with the `function` keyword.
 
-
 ### No `arguments` in arrow function
 
 The `arguments` object does not work in arrow function.
 
 ```js
-const a = () => { console.log(arguments); }
+const a = () => {
+	console.log(arguments);
+};
 a();
 // ReferenceError: arguments is not defined
 ```
+
 Remember I asked why JavaScript decided to have the rest
 syntax and the `arguments` features at the same time when
 they do the same thing?
 The JavaScript language committee decided that `arguments`
-is such a bad idea that it would do the world a great good 
-to disable that syntax in newer language features. As a result, 
-you are forced to use the better rest parameters syntax to use 
+is such a bad idea that it would do the world a great good
+to disable that syntax in newer language features. As a result,
+you are forced to use the better rest parameters syntax to use
 arrow functions at all.
 
-
 ### 🍏Application: React callback revisited
 
 Now we know that the `this` binds statically with arrow function.
 We can replace `bind` with arrow function in `MyComponent`:
 
 ```js
 class MyComponent extends React.Component {
-    constructor() {
-        this.state = { count: 0 };
-        // we removed the bind
-    }
-
-    incrementCounter() {
-        const currentCount = this.state.count;
-        this.setState({ count: currentCount + 1 });
-    }
-
-    render() {
-        const { count } = this.state;
-        return (
-            <div>
-                <h1>count: {count}</h1>
-                {/* We instead use an arrow function here */}
-                <button onClick={() => { this.incrementCounter(); }}> plus 1 </button>
-            </div>
-        );
-    }
+	constructor() {
+		this.state = { count: 0 };
+		// we removed the bind
+	}
+
+	incrementCounter() {
+		const currentCount = this.state.count;
+		this.setState({ count: currentCount + 1 });
+	}
+
+	render() {
+		const { count } = this.state;
+		return (
+			<div>
+				<h1>count: {count}</h1>
+				{/* We instead use an arrow function here */}
+				<button
+					onClick={() => {
+						this.incrementCounter();
```

Comment:
i fear this one might need to be reverted back to original formatting for readability